crates/bevy_ecs/src/bundle.rs

```diff
@@ -667,6 +667,30 @@ impl<'w> BundleInserter<'w> {
         let table = self.table.as_mut();
         let archetype = self.archetype.as_mut();
 
+        // SAFETY: All components in the bundle are guaranteed to exist in the World
+        // as they must be initialized before creating the BundleInfo.
+        unsafe {
+            // SAFETY: Mutable references do not alias and will be dropped after this block
+            let mut deferred_world = self.world.into_deferred();
+
+            deferred_world.trigger_on_replace(
+                archetype,
+                entity,
+                bundle_info
+                    .iter_components()
+                    .filter(|component_id| !add_bundle.added.contains(component_id)),
```

Comment:
Would prefer to avoid the search here if we can. Since the `AddBundle` is cached we can consider pre-computing the replaced ids there.